tap_shopify/streams/base.py

```diff
@@ -50,7 +65,23 @@ def retry_after_wait_gen(**kwargs):
     sleep_time_str = resp.headers.get('Retry-After', resp.headers.get('retry-after'))
     yield math.floor(float(sleep_time_str))
 
+# boolean function to check if the error is 'timeout' error or not
+def is_timeout_error(error_raised):
+    """
+        This function checks whether the error contains 'timed out' substring and return boolean
+        values accordingly, to decide whether to backoff or not.
+    """
+    # retry if the error string contains 'timed out'
+    if str(error_raised).__contains__('timed out'):
+        return False
+    return True
+
 def shopify_error_handling(fnc):
+    @backoff.on_exception(backoff.expo, # timeout error raise by Shopify
+                          (pyactiveresource.connection.Error, socket.timeout),
+                          giveup=is_timeout_error,
+                          max_tries=MAX_RETRIES,
```

Comment:
@harshpatel4crest Don't we have to use `retry_after_wait_gen` function to wait for the next retry?

Reply:
In the code, `retry_after_wait_gen` is used for the request error that contains the header in the error response.

Reply:
Ok, so `retry_after_wait_gen` will be used in case of Errors like 429 (Max retries). Got it, Thank you